### PR TITLE
Add `--binary` flag for `info` command

### PR DESCRIPTION
**Pull request by smiasojed**
Closes https://github.com/paritytech/cargo-contract/issues/800

The combination --binary and --output-json returns JSON: 
```
{ 
  "wasm": "0x..."
}
```
I assumed that binary flag is a behaviour switch, and output-json just formatter.
The another option is to combine wasm hex code with the rest of info command default content:
```
{
  "trie_id": "f8ac805216a9318ab898d0777e24ac52583ca690a218556cfeb6b05744d191cf",
  "code_hash": "0x1a56980585ad4097aaea9daf1bee2332e2a9cf579d0c1a3b91a81900cb399a53",
  "storage_items": 1,
  "storage_item_deposit": 100000000000
  "wasm": "0x..."
}
```
I think that cleaner approach is the first one, where --binary is treated as behaviour switch

### CODE REVIEW

File: crates/extrinsics/src/lib.rs [hunk 1]
```diff
@@ -363,6 +366,23 @@ impl ContractInfo {
     }
 }
 
+/// Fetch the contract wasm code from the storage using the provided client and code hash.
+pub async fn fetch_wasm_code(
+    hash: CodeHash,
+    client: &Client,
+) -> Result<Option<BoundedVec<u8>>> {
```
Comment (ascjones) [inline crates/extrinsics/src/lib.rs:373]: Should convert the `BoundedVec` to a `Vec` for the return type: better not to expose types generated by subxt in the public api of this crate.

File: crates/cargo-contract/src/cmd/info.rs [hunk 2]
```diff
@@ -50,6 +54,9 @@ pub struct InfoCommand {
     /// Export the instantiate output in JSON format.
     #[clap(name = "output-json", long)]
     output_json: bool,
+    /// Display the contract's Wasm bytecode.
```
Comment (ascjones) [inline crates/cargo-contract/src/cmd/info.rs:57]: Should we be more specific here and say "pristine" bytecode, i.e. non-instrumented? We may, for example add the possiblility to retrieve the instrumented byte code.

**Review by ascjones — COMMENTED** [review]
Looks good, just the small nitpick about the return type and a question.

Also please update the description with "Closes https://github.com/paritytech/cargo-contract/issues/800", so we know which issue is being dealt with and it closes automatically when this is merged.

File: crates/cargo-contract/src/cmd/info.rs (hunk 2 above)
Comment (smiasojed) [reply crates/cargo-contract/src/cmd/info.rs:57]: It is named according to Michi comment https://github.com/paritytech/cargo-contract/pull/1013/files/234b258a2b11c309ab18ee09b35996ced1fb0c8e#r1143947838

But if you wish I can change it

File: crates/extrinsics/src/lib.rs (hunk 1 above)
Comment (smiasojed) [reply crates/extrinsics/src/lib.rs]: Done

File: crates/cargo-contract/src/cmd/info.rs (hunk 2 above)
Comment (ascjones) [reply crates/cargo-contract/src/cmd/info.rs:57]: No in that case we can keep it as it is.

**Review by ascjones — APPROVED** [review]
LGTM!

**Pull request merged**